src/main/java/org/wildfly/security/auth/server/MechanismConfiguration.java

```diff
@@ -41,14 +43,19 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class MechanismConfiguration {
+
+    private final MechanismConfiguration parent;
     private final NameRewriter preRealmRewriter;
     private final NameRewriter postRealmRewriter;
     private final NameRewriter finalRewriter;
     private final RealmMapper realmMapper;
     private final Map<String, MechanismRealmConfiguration> mechanismRealms;
     private final List<SecurityFactory<Credential>> serverCredentialFactories;
+    private final Map<String, MechanismConfiguration> serverSpecificConfiguration;
```

Comment:
I went with server name as that is what is used for SASL, in HTTP Host actually has a slightly different definition where it can also include port information which I am not intending to be a part of this mapping.